rclcpp/src/rclcpp/time.cpp

```diff
@@ -278,7 +278,7 @@ Time::operator-=(const rclcpp::Duration & rhs)
 Time
 Time::max()
 {
-  return Time(std::numeric_limits<int32_t>::max(), 999999999);
+  return Time(std::numeric_limits<int64_t>::max());
```

Comment:
Take a look to this change, it seemed to me that the previous value didn't have much sense as it is not near the max value described by header.

Reply:
This came from the original commit back in 2018: https://github.com/ros2/rclcpp/pull/538 .  While I definitely agree with you that this is weird, I think this deserves a separate PR.  Particularly because `rclcpp::Duration` has the same problem.

Reply:
@clalancette I didn't have context about that. Do you think I should open a PR to change that? The operators operating on Time/Duration seem to check for overflow/underflow, I don't see why the max value has to be limited.

If you think it requires further discussion I can open an issue instead of a PR, and for the time being adjust the time to pass with current implementation and reference the issue.

Reply:
> If you think it requires further discussion I can open an issue instead of a PR, and for the time being adjust the time to pass with current implementation and reference the issue.

Yeah, I think it may need further discussion.  So let's revert it here, make the test check for the current behavior (even though it is weird), and open a separate issue/PR to address fixing it.

Reply:
#1351 should follow up, and 7ba92dd changes the test to work with the implementation as is.